include/mata/nfa/nfa.hh

```diff
@@ -280,13 +293,15 @@ public:
      * @return Sequence of @c Trans transitions leading to @p state_to.
      * (!slow!, traverses the entire delta)
      */
+     //this function somehow does not fit with the planned iterator interface, but I don't know what to do
     TransSequence get_transitions_to(State state_to) const;
 
     /**
      * Unify transitions to create a directed graph with at most a single transition between two states.
      * @param[in] abstract_symbol Abstract symbol to use for transitions in digraph.
      * @return An automaton representing a directed graph.
      */
+     //why the get_ everywhere?
     Nfa get_one_letter_aut(Symbol abstract_symbol = 'x') const;
 
```

Comment:
agree, getter should be called `get`.

Reply:
`get` is fine. We should be consistent.

Reply:
Yes, lets be consistent, but how. 
- This one or instance creates a new automaton by modifying every transition of the original, a big computation. Should it then be rather create or something?
- Will we also use get_deterministic, get_minimal, etc? Feels somehow ugly.
- We use those get, make, create, construct without a clear system. It will be hard to agree on a system that makes sense and works in all situations and is liked by everybody. It will keep generating discussions. I think it is much easier to just remove those words whenever no confusion can happen.  Here, one_letter_aut is a clear description of the thing returned by the function, it does not contain noise, nothing to wander about ... ?

Reply:
It is a possible approach, too. It goes against C++ idioms, but other languages implement and encourage @kilohsakul's suggested approach instead. For example, in Rust, you have methods such as `Class::from(<something>)` and `Class::from_something_else(<something_else>` (because you do not have function overloading in Rust for safety reasons), not `Class::construct_from(<something>`. Similarly, you use `Class::new(<initial parameters>)` and `Class::new_special_type_1(<special type one default parameters>)` for constructor functions. There is no `get_`, `make_`, `create_`, etc. It is just a different naming convention.

I am personally fine with either. Note that it goes against C++ idioms, but it is acceptable even in C++, I think. As of know, I believe it is just up to our preferences and taste. I do not believe either option brings any significant advantages that would highly overshadow the other approach.

Reply:
yes it is a matter of taste. I still like my taste better :). My arguments are, shorter, simpler (no need to wander about which of the four words to use), less discussion, less noise. 

The "getters" mentioned above, I thought they are little things that access private data members and do not compute anything much, is that right?

Reply:
That depends. What you are talking about is the classical meaning of a getter, as automatically generated for classes in C#, or manually in Python, for example.

The second meaning is anything that only returns something new from the data in the class, without modifying the class data. The smaller the computation that has to be done in the “getter”, the better. But for functions computing something larger, a different prefix might be more appropriate, such as `compute_`, `make_`, `create_`, etc.

The decision you made in another conversation however resolves this by completely removing all the prefixes. Then there is no confusion possible.

Reply:
I personally agree with everything you said. If we move this method to `builder` as a normal function, its intent should be clear and the name should be easy to agree upon: `Nfa one_letter_nfa(const Nfa& nfa)` or `Nfa one_letter_nfa_from(const Nfa& nfa)`, for example.